third-party/realdds/doc/initialization.md

```diff
@@ -0,0 +1,117 @@
+
+See also: [device](device.md), [notifications](notifications.md)
+
+
+# Initialization
+
+In order to stream, a client must know the stream names, available formats, etc.
+
+The only way for a client to get these is by subscribing to the `notification` topic. When the server detects a subscriber on this topic, it will broadcast initialization messages in the following order:
+
+- `device-header`
+- `device-options` - optional
+- For each stream:
+    - `stream-header`
+    - `stream-options`
+
+These initialization messages should only have effect on devices that are not already initialized. They are expected in the above order.
+
+Once all streams have been received, the device is initialized and ready to use. See [Streaming](streaming.md).
+
+
+#### `device-header`
+
+This is the very first message, for example:
+
+```JSON
+{
+    "id": "device-header",
+    "n-streams": 5,
+    "extrinsics": [
+        ["Depth","Gyro",[1.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,1.0,-0.005520000122487545,0.005100000184029341,0.011739999987185001]]
+        ["Depth","Infrared 2",[1.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,1.0,-0.04986396059393883,0.0,0.0]]
+    ]
+}
+```
+
+Mainly the number of streams to expect. The device will wait for this many stream headers to arrive to finish initialization.
+
+
+#### `device-options`
+
+This is optional: not all devices have options.
+
+```JSON
+{
+    "id": "device-options",
+    "options": [
+        {"description":"Enable/Disable global timestamp","name":"Global Time Enabled","range-default":1.0,"range-max":1.0,"range-min":0.0,"range-step":1.0,"value":1.0},
+        {"description":"HDR Option","name":"Hdr Enabled","range-default":0.0,"range-max":1.0,"range-min":0.0,"range-step":1.0,"value":0.0}
+    ]
+}
+```
+
+Each option is a floating point `value`, with a `name`, `description`, valid range (`default`, `min`, `max`, `step`).
+
+These options will be shown in the Viewer.
+
+
+#### `stream-header`
+
+Information about a specific stream:
+- `name` is the stream name, e.g. `Color`
+    - This will be shown in the Viewer
+    - This is also the name of the topic containing the stream data
+        - E.g., `rt/<topic-root>_Color`
+        - If it contains a space, the space will be replaced by `_` - e.g., `rt/<topic-root>_Infrared_1`
+- `profiles` array
+    - Each profile is itself an array of `[frequency, format, width, height]`
+    - Format is a string representation, similar to the image encoding in ROS
+- `default-profile-index` is the index into the `profiles` for the default profile
+- `sensor-name` is the name of the sensor
+    - Sometimes, a single sensor may have multiple available Streams, e.g. `Depth` may have a `Depth` stream and also `Infrared`
```

Comment:
Since sensor is a librealsense convention I think this should be rephrased a bit. Maybe "Sometimes multiple streams can be grouped together logically. RealSense cameras usually group physically related streams into a `sensor` group"

Reply:
This is from the server implementation point of view: we need to explain why they would be grouped together, and not just that it's a preference type of thing.

Reply:
Changed to:
> - Sometimes, a single sensor may produce multiple streams, e.g. `Stereo Module` produces a `Depth` stream and also `Infrared`